Source/control.cpp

```diff
@@ -1290,7 +1290,8 @@ void DrawInfoBox(const CelOutputBuffer &out)
 					PrintMonstHistory(monster[pcursmonst].MType->mtype);
 				}
 			} else if (pcursitem == -1) {
-				strcpy(infostr, towners[pcursmonst]._tName);
+				std::string_view townerName = towners[pcursmonst]._tName;
+				strncpy(infostr, townerName.data(), townerName.length());

```

Comment:
`strncpy` does not append `\0`, this must be done manually:

```suggestion
				strncpy(infostr, townerName.data(), townerName.length());
				infostr[townerName.length()] = '\0';
```